src/Workspaces/Core/Portable/Rename/ConflictEngine/DeclarationConflictHelpers.cs

```diff
@@ -58,51 +61,43 @@ public bool Equals(List<ITypeSymbol> x, List<ITypeSymbol> y)
                 return x.SequenceEqual(y);
             }
 
-            public int GetHashCode(List<ITypeSymbol> obj)
+            public int GetHashCode(ImmutableArray<ITypeSymbol> obj)
             {
-                // This is a very simple GetHashCode implementation. Doing something "fancier" here
-                // isn't really worth it, since to compute a proper hash we'd end up walking all the
-                // ITypeSymbols anyways.
-                return obj.Count;
+                int hash = 13;
```

Comment:
Arbitrary.  We have a helper for combining hashcodes, but no guidance for where to start.  In any case, this part of the change has been reverted.